File: corpus/typescript/fixture_04.ts
type CartItem4 = { price: number; qty?: number };
export function calculateDiscount4(cartItems: CartItem4[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 40 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook4 {
  lookupPrice(sku: string): number { return sku.length + 4; }
}

File: corpus/go/fixture_04.go
package fixture4

type PriceBook4 struct {
    Multiplier float64
}

func CalculateDiscount4(cartItems []float64, taxRate float64) float64 {
    totalAmount := 0.0
    for _, price := range cartItems {
        totalAmount += price
    }
    if totalAmount > 40.0 {
        return totalAmount * (1.0 + taxRate)
    }
    return totalAmount
}
